httpx/middlewarex/zap_log_formatter.go

```diff
@@ -0,0 +1,54 @@
+package middlewarex
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5/middleware"
+	"go.uber.org/zap"
+)
+
+// ZapLogFormatter is a go-chi/chi/middleware.LogFormatter that
+// writes to a *zap.Logger.
+type ZapLogFormatter struct {
+	Logger *zap.Logger
+}
+
+func (l *ZapLogFormatter) NewLogEntry(r *http.Request) middleware.LogEntry {
+	logger := l.Logger.With(
+		zap.Time("ts_utc", time.Now().UTC()),
+		zap.String("req_id", middleware.GetReqID(r.Context())),
```

Comment:
Already done here: https://github.com/zenledger-io/go-utils/pull/2/files#diff-a09f73cdbb13ea16837cd9d008e6697e189d5ed84ff6ff37f86f7d5a4d61e5abR49

Reply:
> Also it might make sense to use the log package to set and retrieve the id, that way downstream a method could pull the uuid and thus retrieve a logger that sets uuid withjust the context instead of having the whole request

I'm not 100% what you mean. This logger will be made available to all downstream functions, thereby associating all logs with the request.